kuma/core/checks.py

```diff
@@ -0,0 +1,70 @@
+import json
+import os
+try:
+    from json.decoder import JSONDecodeError
+except ImportError:
+    # Still in Python 2
+    JSONDecodeError = ValueError
+try:
+    FileNotFoundError
+except NameError:
+    # Still in Python 2
+    FileNotFoundError = IOError
+
+from django.conf import settings
+from django.core.checks import Error, register as register_check, Warning
+
+
+WARNING_MISSING_I18N_DATA = 'kuma.core.W001'
+ERROR_MISSING_I18N_FILE = 'kuma.core.E001'
+ERROR_CORRUPT_I18N_FILE = 'kuma.core.E002'
+
+
+def react_i18n_check(app_configs, **kwargs):
+    """For every settings.ACCEPTED_LOCALES there's supposed to be one
+    i18n file prepared.
+    This makes sure you have run `make compilejsi18n` and
+    `make compile-react-i18n` before starting Django.
+    """
+    errors = []
+    general_hint = "Run 'make compile-react-i18n'"
+    for locale in settings.ACCEPTED_LOCALES:
+        path = os.path.join(
+            settings.BASE_DIR, 'static', 'jsi18n', locale, 'react.json')
+        try:
+            with open(path) as f:
+                try:
+                    data = json.load(f)
+                except JSONDecodeError as exception:
+                    errors.append(Error(
+                        'Locale file {} is corrupt ({})'.format(
+                            path, exception
+                        ),
+                        hint=general_hint,
+                        id=ERROR_CORRUPT_I18N_FILE,
+                    ))
+                    continue
+
+                missing = [
+                    key for key in ('catalog', 'plural', 'formats')
+                    if key not in data
+                ]
+                if missing:
+                    errors.append(Warning(
+                        'Locale file {} is missing keys {!r}'.format(
+                            path, missing
+                        ),
+                        hint=general_hint,
+                        id=WARNING_MISSING_I18N_DATA,
```

Comment:
Hmm... Your call. " I'd go ahead and treat this case as a data corruption error". So be it. 

Is there anything else you think should only yield a warning? 
To be honest, I just picked anything because I wanted to justify adding an example of `Warning` in this PR. I'd rather you direct more explicitly what exact you think is so busted that it should prevent the startup. 

Perhaps it's entirely silly to do any analysis of the *content* of the JSON files because it's so far away from where it matters. In principle, the checks should be lightweight. We could, for example, just check that the JSON isn't empty or something like that.

Reply:
@davidflanagan I'm ready to make a final call with the code. Please guide me. 
System checks should be lightweight in principle since their business logic is relatively far away from the where the data gets used. 

I'm probably in favor of dropping any `Warning` since it's so likely to not get noticed.